lib/uptime/env.ex

```diff
@@ -27,23 +31,81 @@ defmodule Uptime.Env do
     :ok
   end
 
+  @doc """
+  Read a value from the environment.
+  First check for a secrets file, then the environment, then the config file.
+  secrets file > environment variable > config file
+  """
   @spec read(String.t()) :: String.t()
   def read(name) do
-    case File.read("/etc/secrets/#{name}") do
-      {:ok, value} -> value
-      _not_found -> System.get_env(name)
+    prefixed_name = "#{@prefix}_#{name}"
+
+    secret_file_result =
+      @secrets_directory
+      |> Path.join(prefixed_name)
+      |> File.read()
+
+    case secret_file_result do
+      {:ok, value} ->
+        value
+
+      _not_found ->
+        case read_from_env(prefixed_name) do
+          {:ok, value} ->
+            value
+
+          _not_found ->
+            name
+            |> String.downcase()
+            |> read_from_config_file(find_config_path())
+        end
```

Comment:
crazy idea, but what do you think, then we also make @config_path a module attr that can call the function you have defined at compile time to set the value and make read_from_config_file return an ok tuple

```suggestion
secret_file_path = Path.join(@secrets_directory, prefixed_name)

with {:error, _error} <- File.read(secret_file_path),
  :error <- System.fetch_env(prefixed_name)
  :error <- read_from_config_file(name) do
    raise "Failed to get config value for #{name}"
  else
    {:ok, value} -> 
      if valid_env_var_value(value) do
      {:ok, value}
    else
      raise "Invalid value for #{name}"
    end
end
```